cloud/src/meta-service/txn_lazy_committer.cpp

```diff
@@ -0,0 +1,459 @@
+                    msg_ = ss.str();
+                    LOG(WARNING) << msg_;
+                    break;
+                }
+
+                int64_t table_id = -1;
+                for (auto& [tmp_rowset_key, tmp_rowset_pb] : tmp_rowset_metas) {
+                    if (table_id <= 0) {
+                        table_id = tablet_ids[tmp_rowset_pb.tablet_id()].table_id();
+                    }
+                    txn->remove(tmp_rowset_key);
+                }
+
+                DCHECK(table_id > 0);
+                DCHECK(partition_id > 0);
+
+                std::string ver_val;
+                std::string ver_key =
+                        partition_version_key({instance_id_, db_id, table_id, partition_id});
+                err = txn->get(ver_key, &ver_val);
+                if (TxnErrorCode::TXN_OK != err) {
+                    code_ = err == TxnErrorCode::TXN_KEY_NOT_FOUND
+                                    ? MetaServiceCode::TXN_ID_NOT_FOUND
+                                    : cast_as<ErrCategory::READ>(err);
+                    ss << "failed to get partiton version, txn_id=" << txn_id_
+                       << " key=" << hex(ver_key) << " err=" << err;
+                    msg_ = ss.str();
+                    LOG(WARNING) << msg_;
+                    break;
+                }
+                VersionPB version_pb;
+                if (!version_pb.ParseFromString(ver_val)) {
+                    code_ = MetaServiceCode::PROTOBUF_PARSE_ERR;
+                    ss << "failed to parse version pb txn_id=" << txn_id_
+                       << " key=" << hex(ver_key);
+                    msg_ = ss.str();
+                    break;
+                }
+
+                if (version_pb.has_txn_id() && version_pb.txn_id() == txn_id_) {
+                    version_pb.clear_txn_id();
+                    ver_val.clear();
+                    if (!version_pb.SerializeToString(&ver_val)) {
+                        code_ = MetaServiceCode::PROTOBUF_SERIALIZE_ERR;
+                        ss << "failed to serialize version_pb when saving, txn_id=" << txn_id_;
+                        msg_ = ss.str();
+                        return;
+                    }
+                    txn->put(ver_key, ver_val);
+
+                    err = txn->commit();
+                    if (err != TxnErrorCode::TXN_OK) {
+                        code_ = cast_as<ErrCategory::COMMIT>(err);
+                        ss << "failed to commit kv txn, txn_id=" << txn_id_ << " err=" << err;
+                        msg_ = ss.str();
+                        break;
+                    }
+                }
+            }
+            make_committed_txn_visible(instance_id_, db_id, txn_id_, txn_kv_, code_, msg_);
+        } while (false);
+    } while (code_ == MetaServiceCode::KV_TXN_CONFLICT &&
+             retry_times++ < config::txn_store_retry_times);
+    {
+        std::unique_lock<std::mutex> lock(mutex_);
+        this->finished_ = true;
```

Comment:
do we count it as a finish even if `retry_times > config::txn_store_retry_times`? line 413 may be false and make it reach here.

Reply:
> do we count it as a finish even if `retry_times > config::txn_store_retry_times`? line 413 may be false and make it reach here.

Add more log about code and retry times